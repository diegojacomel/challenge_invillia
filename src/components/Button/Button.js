/** Modules */
import React from 'react'
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ButtonStyle = styled.button`
    display: inline-block;
    border: none;
    cursor: pointer;
    outline: none;
    font-family: 'Open Sans', Arial, sans-serif;
    background: ${props => props.theme.colors[props.color]};
    color: ${props => props.color !== "warning" ? props.theme.colors.white : props.theme.colors.grayDark};
    padding: ${props => props.theme.doubleSpacing[props.padding]};
    border-radius: ${props => props.theme.rounded[props.rounded]};
    font-weight: ${props => props.bold ? "bold" : "normal"};
    font-size: ${props => props.theme.fontSize[props.fontSize]};
`

const ButtonElement = ({ children, color, padding, rounded, bold, fontSize, onClick }) => (
    <ButtonStyle color={color} padding={padding} rounded={rounded} bold={bold} fontSize={fontSize} onClick={onClick}>
        {children}
    </ButtonStyle>
)

function Button({ children, color, padding, rounded, bold, fontSize, onClick }) {
    return (
        <ButtonElement color={color} padding={padding} rounded={rounded} bold={bold} fontSize={fontSize} onClick={onClick} id="testButton">
            {children}
        </ButtonElement>
    )
}

Button.propTypes = {
    color: PropTypes.string,
    padding: PropTypes.string,
    rounded: PropTypes.string,
    bold: PropTypes.bool,
    fontSize: PropTypes.string,
    onClick: PropTypes.func
}

Button.defaultProps = {
    color: "primary",
    padding: "md",
    rounded: "md",
    bold: false,
    fontSize: "sm",
    onClick: () => {}
}

export default Button;