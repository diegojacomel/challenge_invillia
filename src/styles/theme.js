const primaryTheme = {
    colors: {
        primary: '#007bff',
        success: '#28a745',
        secondary: '#6c757d',
        warning: "#ffc107",
        black: '#000000',
        white: '#FFFFFF',
        gray: '#cccccc',
        grayLight: '#F2F2F2',
        grayDark: '#333333'
    },
    fontSize: {
        xs: "14px",
        sm: "16px",
        md: "20px",
        lg: "25px",
        hg: "40px",
        hhg: "60px"
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '80px',
        ggt: '100px',
        hg: '200px',
        hhg: '400px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    },
    doubleSpacing: {
        xs: "2px 5px",
        sm: "3px 10px",
        md: "5px 12px",
        lg: "8px 16px",
        xg: "10px 20px",
        hg: "20px 40px"
    },
    rounded: {
        none: '0',
        xs: '2px',
        sm: '3px',
        md: '5px',
        lg: '10px'
    }
}

export {
    primaryTheme
}