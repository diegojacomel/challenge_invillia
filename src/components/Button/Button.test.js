// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/theme';

// Component
import Button from './Button';

describe('Button', () => {
    const wrapper = shallow(<Button>Text button</Button>);

    test('Render without crashing', () => {
        shallow(<Button>Text button</Button>);
    })

    test('It exist', () => {
        expect(wrapper.find('#testButton')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testButton');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testButton')).toHaveDisplayName('ButtonElement');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testButton')).toMatchSelector('ButtonElement');
    })

    test('Have props', () => {
        expect(wrapper.find('#testButton')).toHaveProp('color');
        expect(wrapper.find('#testButton')).toHaveProp('padding');
        expect(wrapper.find('#testButton')).toHaveProp('rounded');
        expect(wrapper.find('#testButton')).toHaveProp('bold');
        expect(wrapper.find('#testButton')).toHaveProp('fontSize');
        expect(wrapper.find('#testButton')).toHaveProp('onClick');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Button color="primary" padding="md" rounded="md" bold={true} fontSize="md" onClick={() => {}}>
                    Text button
                </Button>
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})