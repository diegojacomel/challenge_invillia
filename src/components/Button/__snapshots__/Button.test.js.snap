// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render correctly 1`] = `
<button
  className="sc-bdVaJa jfsHNJ"
  color="primary"
  fontSize="md"
  onClick={[Function]}
>
  Text button
</button>
`;
